LibreNMS/Authentication/ADAuthorizationAuthorizer.php

```diff
@@ -0,0 +1,354 @@
+            /*
+             * Cache positiv result as this will result in more queries which we
+             * want to speed up.
+             */
+            $this->authLdapSessionCacheSet('user_exists', 1);
+            return 1;
+        }
+
+        return 0;
+    }
+
+
+    public function getUserlevel($username)
+    {
+        $userlevel = $this->authLdapSessionCacheGet('userlevel');
+        if ($userlevel) {
+            return $userlevel;
+        } else {
+            $userlevel = 0;
+        }
+
+        // Find all defined groups $username is in
+        $search = ldap_search(
+            $this->ldap_connection,
+            Config::get('auth_ad_base_dn'),
+            get_auth_ad_user_filter($username),
+            array('memberOf')
+        );
+        $entries = ldap_get_entries($this->ldap_connection, $search);
+
+        // Loop the list and find the highest level
+        foreach ($entries[0]['memberof'] as $entry) {
+            $group_cn = $this->getCn($entry);
+            $auth_ad_groups = Config::get('auth_ad_groups');
+            if ($auth_ad_groups[$group_cn]['level'] > $userlevel) {
+                $userlevel = $auth_ad_groups[$group_cn]['level'];
+            }
+        }
+
+        $this->authLdapSessionCacheSet('userlevel', $userlevel);
+        return $userlevel;
+    }
+
+
+    public function getUserid($username)
+    {
+        $user_id = $this->authLdapSessionCacheGet('userid');
+        if (isset($user_id)) {
+            return $user_id;
+        } else {
+            $user_id = -1;
+        }
+
+        $attributes = array('objectsid');
+        $search = ldap_search(
+            $this->ldap_connection,
+            Config::get('auth_ad_base_dn'),
+            get_auth_ad_user_filter($username),
+            $attributes
+        );
+        $entries = ldap_get_entries($this->ldap_connection, $search);
+
+        if ($entries['count']) {
+            $user_id = preg_replace('/.*-(\d+)$/', '$1', $this->sidFromLdap($entries[0]['objectsid'][0]));
+        }
+
+        $this->authLdapSessionCacheSet('userid', $user_id);
+        return $user_id;
+    }
+
+
+    public function deluser($userid)
+    {
+        dbDelete('bill_perms', '`user_id` =  ?', array($userid));
+        dbDelete('devices_perms', '`user_id` =  ?', array($userid));
+        dbDelete('ports_perms', '`user_id` =  ?', array($userid));
+        dbDelete('users_prefs', '`user_id` =  ?', array($userid));
+        dbDelete('users', '`user_id` =  ?', array($userid));
+        return dbDelete('users', '`user_id` =  ?', array($userid));
+    }
+
+
+    public function getUserlist()
+    {
+        $userlist = array();
+        $userhash = array();
+
+        $ldap_groups = $this->getGroupList();
+
+        foreach ($ldap_groups as $ldap_group) {
+            $search_filter = "(memberOf=$ldap_group)";
+            if (Config::get('auth_ad_user_filter')) {
+                $search_filter = "(&{Config::get('auth_ad_user_filter')}$search_filter)";
```

Comment:
Typo here causes the Config::get to be in the filter instead of the intended output.